compiler/vm/vm.nim

```diff
@@ -73,26 +73,44 @@ const
 
 proc stackTraceImpl(
     c:          TCtx,
-    sframe:     PStackFrame,
+    sframe:     StackFrameIndex,
     pc:         int,
     lineInfo:   TLineInfo,
     infoOrigin: InstantiationInfo,
     recursionLimit: int = 100
   ) =
-
-  proc aux(c: TCtx, sframe: PStackFrame, pc, depth: int, res: var SemReport) =
-    if sframe != nil:
+  ## Generate and report the stack trace starting at frame `sframe` (inclusive)
+  assert c.sframes.len > 0
+
+  # TODO: use the below once exception handling is reworked, and stack frames 
+  # don't form a tree any longer
+  #[
+  let bottom = high(c.sframes)
+  let top = max(0, bottom - recursionLimit)
+  
+  # TODO: Skipped number of entries don't get reported? How is/was this intended to be handled?
```

Comment:
I think the might be a bug, but for a comment on the whole proc it shouldn't be a recursive implementation, it can easily be a loop:

The intention:
- print the most recent frame last
- descend the stack and do so again
- until we run out of stack or reach depth
- if any frames remain, say how many we skipped

The above results in a message such as: "skipped x frames, calls that led up to printing".